lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -371,6 +371,22 @@ func AddInvoice(ctx context.Context, cfg *AddInvoiceConfig,
 
 	}
 
+	// Set our desired invoice features and add them to our list of options.
+	invoiceFeatures := lnwire.NewFeatureVector(
+		lnwire.NewRawFeatureVector(lnwire.TLVOnionPayloadOptional),
+		zpay32.InvoiceFeatures,
+	)
+	options = append(options, zpay32.Features(invoiceFeatures))
+
+	// Generate and set a random payment address for this invoice. If the
+	// sender understands payment addresses, this can be used to avoid
+	// intermediaries probing the receiver.
+	var paymentAddr [32]byte
+	if _, err := rand.Read(paymentAddr[:]); err != nil {
+		return nil, nil, err
+	}
+	options = append(options, zpay32.PaymentAddr(&paymentAddr))
```

Comment:
I think we also need to set it on `channeldb.Invoice` and serialize separately.

Reply:
done